File: app/src/main/java/com/andraganoid/verymuchtodo/auth/AuthBaseFragment.java
package com.andraganoid.verymuchtodo.auth;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;

import com.andraganoid.verymuchtodo.MainActivity;
import com.andraganoid.verymuchtodo.MainViewModel;


public class AuthBaseFragment extends Fragment {

    protected MainViewModel mainViewModel;
    protected MainActivity main;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        main = (MainActivity) getActivity();
        mainViewModel = ViewModelProviders.of(main).get(MainViewModel.class);
    }

}

File: app/src/main/java/com/andraganoid/verymuchtodo/auth/AuthBaseFragment.kt
package com.andraganoid.verymuchtodo.auth

import android.os.Bundle
import androidx.fragment.app.Fragment
import com.andraganoid.verymuchtodo.shortVersion.main.MainActivity
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.ObsoleteCoroutinesApi

@ExperimentalCoroutinesApi
@ObsoleteCoroutinesApi
open class AuthBaseFragment : Fragment() {

    lateinit var main: MainActivity

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        main = activity as MainActivity
    }

    fun loaderVisibility(visibility: Boolean) {
        main.loaderVisibility(visibility)
    }

    fun hideKeyboard() {
        main.hideKeyboard()
    }

    fun showMessage(message: Any?) {
        main.showMessage(message)
    }

//    fun sendEmailToAdmin() {
//        main.sendEmailToAdmin()
//    }

}
